tests/history_test.go

```diff
@@ -702,17 +703,24 @@ func assertStreamConsistency(t testing.TB, clients []*redis.Client, stream strin
 	}
 }
 
-func processCheckResult(t *testing.T, client *utils.Icinga2Client, hostname string, status int) time.Time {
+// processCheckResult sends a passive check result to the Icinga2 API and returns the time of the check result.
+// The provided hostOrServiceName must be either a host name or a service name in the format "host!service".
+func processCheckResult(t *testing.T, client *utils.Icinga2Client, hostOrServiceName string, status int) time.Time {
 	// Ensure that check results have distinct timestamps in millisecond resolution.
 	time.Sleep(10 * time.Millisecond)
 
 	output := utils.RandomString(8)
-	reqBody, err := json.Marshal(ActionsProcessCheckResultRequest{
-		Type:         "Host",
-		Filter:       fmt.Sprintf(`host.name==%q`, hostname),
-		ExitStatus:   status,
-		PluginOutput: output,
-	})
+	aprr := ActionsProcessCheckResultRequest{ExitStatus: status, PluginOutput: output}
```

Comment:
I presume `aprr` is supposed to be an abbreviation of the type name, but then it misses a `c`. However, unless it was a common abbreviation, I don't see how this provides any benefit over a name like let's say `r`. A more meaningful name could be `req` or maybe `processCR` (as we use `cr` frequently to abbreviate check result, at least in the C++ code).